packages/ai/src/authorizers/federated-connections/FederatedConnectionAuthorizerBase.ts

```diff
@@ -128,24 +145,59 @@ export class FederatedConnectionAuthorizerBase<ToolExecuteArgs extends any[]> {
 
     const { connection } = store;
 
-    const subjectToken = await this.getRefreshToken(...toolContext);
+    // Determine which token type to use and get the appropriate subject token
+    let subjectToken: string | undefined;
+    let subjectTokenType: string;
+    const clientId: string =
+      this.auth0.clientId || this.auth0.resourceServerClientId!;
```

Comment:
Shouldn't this also depend on whether a refresh token or an access token was provided?